services/pixiv.py

```diff
@@ -75,15 +76,16 @@ def authentication(self) -> bool:
             try:
                 self.pixivApi.auth(refresh_token=self.token)
                 success = True
+                self.logger.debug("Authentication success!")
                 break
             except pixiv.utils.PixivError:
-                print('The token has expired and needs to be reset!')
-                success = self.refreshToken()
+                self.logger.debug("The token has expired and needs to be reset!")
+                success = self.refresh_token()
```

Comment:
Add lines:
```
      if success:
          break  # exit the while loop if refresh_token was successful
  else:
      raise
```
This is to prevent excessive iterations

Reply:
When we refresh the token, we have to re-authenticate it, so we can't break it, we have to break it in the try section.

Reply:
pixivTokenApi and pixivApi are two different sessions. 
The pixivTokenApi is only used to get the token
The pixivApi is used to log in to pixiv to get illustrations.